frontend/src/metabase/admin/permissions/selectors/data-permissions/fields.ts

```diff
@@ -53,21 +55,40 @@ const buildAccessPermission = (
     entityId,
     DataPermission.VIEW_DATA,
   );
+
   const defaultGroupValue = getFieldsPermission(
     permissions,
     defaultGroup.id,
     entityId,
     DataPermission.VIEW_DATA,
   );
 
-  const warning = getPermissionWarning(
+  const dbValue = getSchemasPermission(
+    originalPermissions,
+    groupId,
+    entityId,
+    DataPermission.VIEW_DATA,
+  );
+
+  const schemaValue = getTablesPermission(
+    originalPermissions,
+    groupId,
+    entityId,
+    DataPermission.VIEW_DATA,
+  );
```

Comment:
This works properly, but this code is super confusing. 

Why do we get the db permission value by checking schema permissions?
Why do we get the schema permission value by checking table permissions?

Reply:
Welcome to the permissions FE code 😬 I didn't name these functions and I have been holding out that I would one day understand why it was done this way... but I haven't figured it out. Schema means databases, tables means schemas, fields means tables.

It is confusing, but it is widespread at this point. I'll go about renaming the functions in a subsequent PR if that's alright with you?

Reply:
I don't know how safe/hard/helpful such a rename would be, so i'll leave that up to your judgment, but we probably do need more comments like this in the permissions code

```ts
// 🥴 
```